pylint/checkers/classes.py

```diff
@@ -936,45 +936,59 @@ def _check_protected_attribute_access(self, node):
         '''
         attrname = node.attrname
 
-        if (is_attr_protected(attrname) and
-                attrname not in self.config.exclude_protected):
+        if not is_attr_protected(attrname) or attrname in self.config.exclude_protected:
```

Comment:
What I don't like about this change is that it conflates a refactoring with the actual change, which impedes the review. Can you try to separate them, so that it would be more obvious what's the new change?